src/redux/reducers/main.ts

```diff
@@ -444,6 +461,7 @@ export const mainSlice = createSlice({
         state.previewLoader.isLoading = false;
         state.previewLoader.targetResourceId = undefined;
         resetSelectionHistory(state, {initialResourceIds: [state.previewResourceId]});
+        state.selectedResourceId = undefined;
       })
       .addCase(previewCluster.rejected, state => {
```

Comment:
don't we need to add the above calls for exiting/rejecting cluster preview also?